dependency/retrieval/main_test.go

```diff
@@ -0,0 +1,100 @@
+package main_test
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/joshuatcasey/libdependency/versionology"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/paketo-buildpacks/occam/matchers"
+	"github.com/paketo-buildpacks/pipenv/retrieval"
+)
+
+func testRetrieval(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+	)
+
+	context("GenerateMetadata", func() {
+		it("will generate metadata for a valid version", func() {
+			metadata, err := main.GenerateMetadata(versionology.NewSimpleVersionFetcher(semver.MustParse("2.4.4")))
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(metadata).To(ConsistOf(versionology.Dependency{
+				ConfigMetadataDependency: cargo.ConfigMetadataDependency{
+					Checksum:       "sha256:c252c2a2219956f88089ffc242b42c8cb9300a368fd3890d63940e4fc9652345",
+					CPE:            "cpe:2.3:a:getcomposer:composer:2.4.4:*:*:*:*:python:*:*",
+					PURL:           "pkg:generic/composer@2.4.4?checksum=c252c2a2219956f88089ffc242b42c8cb9300a368fd3890d63940e4fc9652345&download_url=https://getcomposer.org/download/2.4.4/composer.phar",
+					ID:             "composer",
+					Licenses:       []interface{}{"MIT"},
+					Name:           "composer",
+					Source:         "https://getcomposer.org/download/2.4.4/composer.phar",
+					SourceChecksum: "sha256:c252c2a2219956f88089ffc242b42c8cb9300a368fd3890d63940e4fc9652345",
+					Stacks:         []string{"*"},
+					URI:            "https://getcomposer.org/download/2.4.4/composer.phar",
+					Version:        "2.4.4",
+				},
+				SemverVersion: semver.MustParse("2.4.4"),
+				Target:        "NONE",
+			}))
+		})
+	})
+
+	context("PharDecompress", func() {
+		var (
+			pharPath, destination string
+
+			err error
+		)
+
+		it.Before(func() {
+			pharPath = filepath.Join("testdata", "phar", "composer-2.4.4.phar")
+			destination = t.TempDir()
+		})
+
+		it("will decompress a phar", func() {
+			var pharBytes []byte
+
+			pharBytes, err = os.ReadFile(pharPath)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = main.PharDecompress(bytes.NewReader(pharBytes), destination)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(filepath.Join(destination, "LICENSE")).
+				To(matchers.BeAFileMatching(ContainSubstring("Copyright (c) Nils Adermann, Jordi Boggiano")))
+		})
+	})
+}
+
+func ExamplePharDecompress_printsNothingOnSuccess() {
```

Comment:
What's the point of this function?

Reply:
`phar extract -f` is really verbose. This `Example` function demonstrates that it logs nothing to `STDOUT` on success.

I tried to implement a negative version of this (showing that `phar extract -f` output was sent to `STDOUT` on failure) but it contained dynamic output and was therefore not workable as an `Example` test. It could be tested as a unit test, but that would require injecting an `io.Writer` and a bunch of other changes that I wasn't prepared to do.

Reply:
I see. Are you using it anywhere? I don't think so but might have missed it. If not, I don't think we need this here. A comment will suffice with the behaviour of `phar extract -f`

Reply:
It's an [example test](https://go.dev/blog/examples) so it's run as part of `go test`.

Note that this does not apply when using [`go test -run Unit`](https://github.com/paketo-buildpacks/github-config/blob/main/implementation/scripts/unit.sh#L53), which is why the Library `./scripts/unit.sh` includes [`go test -run Unit|Example`](https://github.com/paketo-buildpacks/github-config/blob/main/library/scripts/unit.sh#L53).

Reply:
Cool, I haven't seen that before. Thanks for the explainer.